### DISCUSSION TITLE
Support for ibm-semeru-runtimes Docker images

### DISCUSSION THREAD

**Discussion by Athou** [renovatebot/renovate#30333] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi,

I recently switched from the [Eclipse Temurin](https://hub.docker.com/_/eclipse-temurin) JRE to [IBM Semeru OpenJ9](https://hub.docker.com/_/ibm-semeru-runtimes) for the [CommaFeed](https://github.com/Athou/commafeed) Docker image.
[Since then](https://github.com/Athou/commafeed/commit/9134f36d3b6d1146e8d8b4d012dde503a0a62bae), Renovate is no longer suggesting updates for the [FROM](https://github.com/Athou/commafeed/blob/master/Dockerfile#L1) tag.

Would it be possible to add support for the Semeru JRE? It would also be awesome if support was also added to the [Java LTS workaround](https://github.com/renovatebot/renovate/blob/cdd745767340c2ee167448ee83aa34a86c9067b9/lib/config/presets/internal/workarounds.ts#L129) so that Renovate would only suggest LTS upgrades.

Thanks for your help!

### Logs (if relevant)

<details><summary>Logs</summary>

```

    "dockerfile": [
      {
        "deps": [
          {
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "currentValue": "open-21-jre",
            "datasource": "docker",
            "depName": "ibm-semeru-runtimes",
            "depType": "final",
            "packageName": "ibm-semeru-runtimes",
            "replaceString": "ibm-semeru-runtimes:open-21-jre",
            "skipReason": "invalid-value",
            "versioning": "docker",
            "warnings": [],
            "updates": []
          }
        ],
        "packageFile": "Dockerfile"
      }
    ],

...
DEBUG: Dependency ibm-semeru-runtimes has unsupported/unversioned value open-21-jre (versioning=docker)
DEBUG: Skipping ibm-semeru-runtimes because no currentDigest or pinDigests
```
</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
You will need to [confugure versioning](https://docs.renovatebot.com/modules/versioning/#why-you-might-need-to-manually-configure-versioning) for this package.

**Reply by Athou** [reply]
Thanks! This seems to be working:

    {
      "matchDatasources": "docker",
      "matchPackageNames": "ibm-semeru-runtimes",
      "versioning": "regex:^open-(?<major>\\d+)?(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?([\\._+](?<build>(\\d\\.?)+))?(-(?<compatibility>.*))?$",
      "allowedVersions": "/^open-(?:8|11|17|21)(?:\\.|-|$)/"
    }

Do I need to maintain `allowedVersions` myself or is there a way to retrieve the value from the [LTS workaround](https://github.com/renovatebot/renovate/blob/cdd745767340c2ee167448ee83aa34a86c9067b9/lib/config/presets/internal/workarounds.ts#L133)? I had to add `open-` at the start of the regex though.

**Reply by RahulGautamSingh** [reply]
You'll have to maintain it.

**Comment by MPV** [comment]
On official/built-in support for detecting Semeru runtimes using the `java-version` datasource in Renovate, I raised this discussion for that now:
- https://github.com/renovatebot/renovate/discussions/34440
